src/main/scala/com/mongodb/spark/rdd/partitioner/BsonValueOrdering.scala

```diff
@@ -191,7 +191,7 @@ trait BsonValueOrdering extends Ordering[BsonValue] {
       case (BsonType.SYMBOL, BsonType.SYMBOL) => (x.asSymbol.getSymbol, y.asSymbol.getSymbol)
       case _                                  => throw new UnsupportedOperationException(s"Cannot compare $x with $y. Not string compatible types.")
     }
-    compareBytes(xString.getBytes("utf-8") zip yString.getBytes("utf-8"))
```

Comment:
The issue here was the missing length check - zip normalizes the length.  This had the impact with some of the partitioning code when sorting string values it could cause ranges to cover duplicate data.